titiler/api/endpoints/mosaic.py

```diff
@@ -0,0 +1,245 @@
+"""API for MosaicJSON Dataset."""
+import asyncio
+import os
+import random
+from functools import partial
+from io import BytesIO
+from typing import Callable, Sequence
+
+import mercantile
+import morecantile
+import numpy
+from cogeo_mosaic.backends import MosaicBackend
+from cogeo_mosaic.mosaic import MosaicJSON
+from cogeo_mosaic.utils import get_footprints
+from rasterio.transform import from_bounds
+from rio_tiler.io.cogeo import tile as cogeoTiler
+from rio_tiler.profiles import img_profiles
+from rio_tiler.utils import render
+
+from titiler.api.deps import CommonTileParams, TileMatrixSetNames
+from titiler.api.endpoints.cog import cog_info, tile_response_codes
+from titiler.api.utils import postprocess
+from titiler.models.metadata import cogBounds, cogInfo
+from titiler.models.mosaic import CreateMosaicJSON, UpdateMosaicJSON
+from titiler.ressources.common import drivers
+from titiler.ressources.enums import ImageMimeTypes, ImageType, PixelSelectionMethod
+from titiler.ressources.responses import ImgResponse
+
+from fastapi import APIRouter, Depends, Path, Query
+from fastapi.routing import APIRoute
+
+from starlette.concurrency import run_in_threadpool
+from starlette.exceptions import HTTPException
+from starlette.requests import Request
+from starlette.responses import Response
+
+
+def _chunks(my_list: Sequence, chuck_size: int):
+    """Yield successive n-sized chunks from l."""
+    for i in range(0, len(my_list), chuck_size):
+        yield my_list[i : i + chuck_size]
```

Comment:
can we use https://github.com/cogeotiff/rio-tiler/blob/cb0a709144b4589bb9940e56716eb741a53f71fd/rio_tiler/utils.py#L23-L26 ?

Reply:
This actually shouldn't be here (I forgot to delete), since we are using `asyncio.Semphore` instead of using `_chunks` to limit concurrency.  I actually didn't realize this helper function was defined in `rio_tiler`, if it is needed in the future projects I will import from there.